cpp/src/cylon/ctx/cylon_context.cpp

```diff
@@ -22,16 +22,16 @@
 
 namespace cylon {
 
-CylonContext *CylonContext::Init() {
-  return new CylonContext(false);
+shared_ptr<CylonContext> CylonContext::Init() {
+  return make_shared<CylonContext>(false);
 }
 CylonContext::CylonContext(bool distributed) {
   this->is_distributed = distributed;
 }
 
-CylonContext *CylonContext::InitDistributed(net::CommConfig *config) {
+shared_ptr<CylonContext> CylonContext::InitDistributed(net::CommConfig *config) {
```

Comment:
I think we need to do the same `shared_ptr` change for `net::CommConfig` as well. @supunkamburugamuve WDYT?

Reply:
There are many places, which is not consistent. I think it is better to take a look at the whole source and decide them and refactor now. In this PR or another. I can help.